GoodreadsEDA/scp_bks.py

```diff
@@ -104,97 +78,124 @@ def print_pg_number(page_count):
-    create_review_table(c)
-
-    for entry in book_urls:
+    # escapes all non-alphanumeric characters for consistent insertion
+    user = re.escape(user)
+    try:
+        c.execute("""INSERT INTO Reviews (book_auth, book_title, book_url, review_score, user_name, review_date)
+                     VALUES (?, ?, ?, ?, ?, ?)""", (author, title, url, score, user, date))
+    except sqlite3.OperationalError:
+        # this should actually never happen now, but I'm leaving it here for good measure
+        traceback.print_exc()
+        c.execute("""INSERT INTO Reviews (book_auth, book_title, book_url, review_score, user_name, review_date)
+                     VALUES (?, ?, ?, ?, 'invalid username', ?)""", (author, title, url, score, date))
+
+
+def print_scraping_status(current_book, total_books, start_timer):
+    """Prints the status of the scraping if the process has progressed an arbitrary amount of books (currently 5)"""
+
+    if current_book % 5 == 0 or current_book == 1:
+        percent_completed = round(current_book / len(total_books) * 100, 2)
+        minutes_elapsed = round((time.perf_counter() - start_timer) / 60, 2)
+        print("-" * 75)
+        print("{}% Complete".format(percent_completed))
+        print("{} minutes elapsed, {} minutes of scraping remaining".format(
+            minutes_elapsed, round(minutes_elapsed * (100.0 - percent_completed), 2)
+        ))
+        print("-" * 75)
+
+
+def scrape_book_info(book_urls, driver, cursor):
+    """Navigates the webdriver to the urls provided and scrapes up to 300 rating results from 10 pages of rating results
+
+    Args:
+        book_urls ([[str, str, str]]): A list of list of strings consisting of book author, book title, and book url
+          to visit to serve as the base scraping url.
+        driver (webdriver): The selenium chromedriver object to visit the urls with.
+        cursor (sqlite3.cursor): A sqlite cursor to pass to methods writing to the DB.
+    """
+
+    scraping_start_timer = time.perf_counter()
+
+    for book_number, entry in enumerate(book_urls):
         author = entry[0]
         title = entry[1]
-        if '"' in title:
-            title = replace_double_quotes(title)
         url = entry[2]
 
-        print_bk_info(author, title, url)
-
-        driver = webdriver.Chrome(chrome_options=options)
         driver.get(url)
 
-        page_count = 1
         # Goodreads keeps a collection of 10 pages of 30 reviews available on the
         # book's page at a time.
-        for x in range(0, 10):
-            print_pg_number(page_count)
+        print("Scraping 10 pages of reviews for \"{}\". Scraping page ".format(title), end="", flush=True)
+        for page_count in range(1, 11):
+            # print_pg_number(page_count)
+            # each set of 10 pages is printed on the same line to save space & cut down on noise
+            if page_count != 10:
+                print("{}..".format(page_count), end="", flush=True)
```

Comment:
These print parameters just make it so subsequent prints are on the same line (`end="\n"` by default)